### PR TITLE
Fix user agent strings and the otel uptime version label.

### PR DESCRIPTION
**Pull request by igorpeshansky**
Update tests to exercise otel on linux.
Also synchronize error behavior between otel and collectd, and some cleanup of test inputs.

### CODE REVIEW

File: confgenerator/confgenerator.go [hunk 1]
```diff
@@ -271,10 +272,25 @@ func defaultStackdriverOutputs() (stackdrivers []*conf.Stackdriver) {
 	}
 }
 
-var userAgentTemplate = template.Must(template.New("useragent").Parse(`{{.Prefix}}/{{.AgentVersion}} (BuildDistro={{.BuildDistro}};Platform={{.Platform}};ShortName={{.ShortName}};ShortVersion={{.ShortVersion}},gzip(gfe))`))
+var versionLabelTemplate = template.Must(template.New("versionlabel").Parse(`{{.Prefix}}/{{.AgentVersion}}-{{.BuildDistro}}`))
+var userAgentTemplate = template.Must(template.New("useragent").Parse(`{{.Prefix}}/{{.AgentVersion}} (BuildDistro={{.BuildDistro}};Platform={{.Platform}};ShortName={{.ShortName}};ShortVersion={{.ShortVersion}})`))
+
+func getVersionLabel(prefix string, hostInfo *host.InfoStat) (string, error) {
```
Comment (qingling128) [inline confgenerator/confgenerator.go:278]: The `getVersionLabel` and `getUserAgent` functions are quite similar. Does it make sense to combine them?

File: confgenerator/testdata/invalid/linux/metrics-receiver_unsupported_type/golden_error [hunk 2]
```diff
@@ -1 +1 @@
-metrics receiver "receiver_1" with type "unsupported_type" is not supported. Supported metrics receiver types: [hostmetrics].
\ No newline at end of file
+metrics receiver "receiver_1" with type "unsupported_type" is not supported. Supported metrics receiver types: [hostmetrics, iis, mssql].
```
Comment (qingling128) [inline confgenerator/testdata/invalid/linux/metrics-receiver_unsupported_type/golden_error:1]: On Linux, `iis` and `mssql` are not supported, right?

File: otel/conf.go [hunk 3]
```diff
@@ -54,7 +55,7 @@ service:
 {{- end}}
 {{define "hostmetrics" -}}
   hostmetrics/{{.HostMetricsID}}:
-    collection_interval: {{.CollectionInterval | validateCollectionInterval "hostmetrics"}}
```
Comment (qingling128) [inline otel/conf.go:57]: Nice catch! Is this worth a separate PR though? Seems a bit unrelated to `Fix user agent strings and the otel uptime version label.`, right?

Comment (igorpeshansky) [reply otel/conf.go:57]: Ah. This is part of synchronizing the error messages across collectd and otel… I'm happy to mention that in the PR description if needed.

**Comment by igorpeshansky** [comment]
As agreed offline, I've split out the first two commits of this PR into #94, and will rebase once that one is merged.

**Comment by qingling128** [comment]
https://github.com/GoogleCloudPlatform/ops-agent/pull/94 was merged. Guess we could rebase this one then. Let me know when it's ready for review.

**Comment by igorpeshansky** [comment]
Rebased. Ready for review.

File: confgenerator/testdata/invalid/linux/metrics-receiver_unsupported_type/golden_error (hunk 2 above)
Comment (qingling128) [reply confgenerator/testdata/invalid/linux/metrics-receiver_unsupported_type/golden_error:1]: Ah I see. This changed because we switched from collectd to otel, which has a TODO. Got it.

**Review by qingling128 — APPROVED** [review]
LGTM
